### PR TITLE
fix(api): fix error when name field omitted in createUser

### PR DESCRIPTION
**Pull request by wreality**
ref #103

### CODE REVIEW

File: backend/database/migrations/2021_01_22_061707_allow_user_name_null.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->string('name')->nullable(false)->change();
```
Comment (gmeben) [inline backend/database/migrations/2021_01_22_061707_allow_user_name_null.php:29]: How does this work? 

If the `down()` methods of all the migrations are ran chronologically, the `users` table won't exist to be modified by the time this `down()` method is ran. Doesn't that make this redundant?

Or is it possible that this `down()` method can be ran independent of the others? (new to me)

File: backend/tests/Feature/CreateUserEndpointTest.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+use Nuwave\Lighthouse\Testing\MakesGraphQLRequests;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class CreateUserEndpointTest extends TestCase
+{
+    use MakesGraphQLRequests;
+    use RefreshDatabase;
+    /**
+     * A basic feature test example.
```
Comment (gmeben) [inline backend/tests/Feature/CreateUserEndpointTest.php:14]: This comment is too generic and the function name is already descriptive enough of what it does.
```suggestion

```

File: backend/tests/Feature/CreateUserEndpointTest.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                    username
+                }
+            }
+        ');
+
+        $response->assertJsonPath("data.createUser.username", "testusername");
+
+
+    }
+
```
Comment (gmeben) [inline backend/tests/Feature/CreateUserEndpointTest.php:33]: Add the missing DocBlock here:
```suggestion
    /**
     * @return void
     */
```

File: backend/tests/Feature/CreateUserEndpointTest.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+namespace Tests\Feature;
+
+use Nuwave\Lighthouse\Testing\MakesGraphQLRequests;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class CreateUserEndpointTest extends TestCase
+{
+    use MakesGraphQLRequests;
+    use RefreshDatabase;
```
Comment (gmeben) [inline backend/tests/Feature/CreateUserEndpointTest.php:12]: I like having a little separation between functions/classes and use statements. This convention seems to be demonstrated everywhere else in the code too.
```suggestion
    use RefreshDatabase;

```

File: backend/tests/Feature/CreateUserEndpointTest.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        
+            mutation {
+                createUser(user: {email: "brandnew@gmail.com", password: "KajSu8viptUrz&", username: "testusername"}) {
+                    username
+                }
+            }
+        ');
+
+        $response->assertJsonPath("data.createUser.username", "testusername");
+
```
Comment (gmeben) [inline backend/tests/Feature/CreateUserEndpointTest.php:46]: Might be useful to also assert that `data.createUser.name` is null. I'll see if I can come up with something.

File: backend/tests/Feature/CreateUserEndpointTest.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        
+            mutation {
+                createUser(user: {name: "", email: "brandnew@gmail.com", password: "KajSu8viptUrz&", username: "testusername"}) {
+                    username
+                }
+            }
+        ');
+
+        $response->assertJsonPath("data.createUser.username", "testusername");
+
```
Comment (gmeben) [inline backend/tests/Feature/CreateUserEndpointTest.php:30]: Might be useful to also assert that `data.createUser.name` is empty. I'll see if I can come up with something.

**Review by gmeben — CHANGES_REQUESTED** [review]

File: backend/database/migrations/2021_01_22_061707_allow_user_name_null.php (hunk 1 above)
Comment (wreality) [reply backend/database/migrations/2021_01_22_061707_allow_user_name_null.php:29]: The down methods would be run in reverse chronological order. It's the undo of up. Also you can just undo a single migration (artisan migrate:rollback --step=1)

File: backend/tests/Feature/CreateUserEndpointTest.php (hunk 2 above)
Comment (wreality) [reply backend/tests/Feature/CreateUserEndpointTest.php:14]: Yep that's the default, thanks for catching it!

File: backend/tests/Feature/CreateUserEndpointTest.php (hunk 6 above)
Comment (gmeben) [reply backend/tests/Feature/CreateUserEndpointTest.php]: I added an assertion to check that the value of this property is an empty string. Let me know what you think.

File: backend/tests/Feature/CreateUserEndpointTest.php (hunk 5 above)
Comment (gmeben) [reply backend/tests/Feature/CreateUserEndpointTest.php]: I added an assertion to check that the value of this property is null. Let me know what you think.

File: backend/tests/Feature/CreateUserEndpointTest.php (hunk 4 above)
Comment (wreality) [reply backend/tests/Feature/CreateUserEndpointTest.php]: Added in a later commit.

File: backend/tests/Feature/CreateUserEndpointTest.php (hunk 3 above)
Comment (wreality) [reply backend/tests/Feature/CreateUserEndpointTest.php]: Added.

File: backend/tests/Feature/CreateUserEndpointTest.php (hunk 2 above)
Comment (wreality) [reply backend/tests/Feature/CreateUserEndpointTest.php]: Added.

File: backend/tests/Feature/CreateUserEndpointTest.php (hunk 4 above)
Comment (wreality) [reply backend/tests/Feature/CreateUserEndpointTest.php]: Resolved

**Review by gmeben — APPROVED** [review]

**Pull request merged**